src/pgduckdb_types.cpp

```diff
@@ -192,6 +169,269 @@ ConvertNumeric(T value, idx_t scale) {
 	return result;
 }
 
+static Datum
+ConvertNumericDatum(const duckdb::Value &value) {
+	if (value.type().id() == duckdb::LogicalTypeId::DOUBLE) {
+		return ConvertDoubleDatum(value);
+	}
+	NumericVar numeric_var;
+	D_ASSERT(value.type().id() == duckdb::LogicalTypeId::DECIMAL ||
+	         value.type().id() == duckdb::LogicalTypeId::HUGEINT ||
+	         value.type().id() == duckdb::LogicalTypeId::UBIGINT);
+	auto physical_type = value.type().InternalType();
+	const bool is_decimal = value.type().id() == duckdb::LogicalTypeId::DECIMAL;
+	uint8_t scale = is_decimal ? duckdb::DecimalType::GetScale(value.type()) : 0;
+
+	switch (physical_type) {
+	case duckdb::PhysicalType::INT16: {
+		numeric_var = ConvertNumeric<int16_t>(value.GetValueUnsafe<int16_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::INT32: {
+		numeric_var = ConvertNumeric<int32_t>(value.GetValueUnsafe<int32_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::INT64: {
+		numeric_var = ConvertNumeric<int64_t>(value.GetValueUnsafe<int64_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::UINT64: {
+		numeric_var = ConvertNumeric<uint64_t>(value.GetValueUnsafe<uint64_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::INT128: {
+		numeric_var = ConvertNumeric<hugeint_t, DecimalConversionHugeint>(value.GetValueUnsafe<hugeint_t>(), scale);
+		break;
+	}
+	default: {
+		throw duckdb::InvalidInputException(
+		    "(PGDuckDB/ConvertDuckToPostgresValue) Unrecognized physical type for DECIMAL value");
+	}
+	}
+	auto numeric = CreateNumeric(numeric_var, NULL);
+	auto datum = NumericGetDatum(numeric);
+	return datum;
+}
+
+static Datum
+ConvertUUIDDatum(const duckdb::Value &value) {
+	D_ASSERT(value.type().id() == duckdb::LogicalTypeId::UUID);
+	D_ASSERT(value.type().InternalType() == duckdb::PhysicalType::INT128);
+	auto duckdb_uuid = value.GetValue<hugeint_t>();
+	pg_uuid_t *postgres_uuid = (pg_uuid_t *)palloc(sizeof(pg_uuid_t));
+
+	duckdb_uuid.upper ^= (uint64_t(1) << 63);
+	// Convert duckdb_uuid to bytes and store in postgres_uuid.data
+	uint8_t *uuid_bytes = (uint8_t *)&duckdb_uuid;
+
+	for (int i = 0; i < UUID_LEN; ++i) {
+		postgres_uuid->data[i] = uuid_bytes[UUID_LEN - 1 - i];
+	}
+
+	return UUIDPGetDatum(postgres_uuid);
+}
+
+template <int32_t OID>
+struct PostgresTypeTraits;
+
+// Specializations for each type
+// BOOL type
+template <>
+struct PostgresTypeTraits<BOOLOID> {
+	static constexpr int16_t typlen = 1;
+	static constexpr bool typbyval = true;
+	static constexpr char typalign = 'c';
+
+	static Datum
```

Comment:
Let's mark all these `ToDatum` functions as `inline`. They are so tiny that there's no need for introducing calling overhead.

Reply:
Let's do the same for some of the smaller `Convert` functions.

Reply:
@JelteF Updated. Thanks for your review. PTAL